core/src/kura.rs

```diff
@@ -335,6 +352,14 @@ impl Kura {
             .lock()
             .push((block.hash(), Some(Arc::new(block))));
     }
+
+    /// Replace the block in kura's in memory block store. Kura will write the block
+    /// to disk in due time.
+    pub fn replace_top_block_blocking(&self, block: VersionedCommittedBlock) {
```

Comment:
```suggestion
    pub fn replace_top_block(&self, block: VersionedCommittedBlock) {
```

The blocking property might change, and we don't really have a non-blocking alternative.